chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1030,7 +1030,11 @@ where
                     // support/#85 for a use case.
                     Err(EthereumContractCallError::Revert("empty response".into()))
                 } else {
-                    call.function.decode_output(&output).map_err(From::from)
+                    // Decode failures are reverts. The reasoning is that if Solidity fails to
+                    // decode an argument, that's a revert, so the same goes for the output.
```

Comment:
I'd say it's a little different with the outputs. Us being able to not decode the output can just mean that there is an ABI mismatch and the contract returns different values/types from the call. I could be wrong?

Reply:
I still think this is the right thing to do. Just not sure the explanation is 100% accurate and won't confuse us later.

Reply:
> Us being able to not decode the output can just mean that there is an ABI mismatch and the contract returns different values/types from the call

How is that different from the input decoding? Both are caused either by an ABI mismatch or by bad data.

Reply:
Oh I see, we encode the inputs and they should always be valid, so that's actually a real error, yhea what I wrote doesn't make sense, I'll change this.

Reply:
We concluded this analogy is pretty much correct as it is.